include/drivers/clock_control/stm32_clock_control.h

```diff
@@ -164,12 +164,20 @@
 #define STM32_CSI_FREQ		0
 #endif
 
-#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(clk_lsi), fixed_clock, okay)
-#define STM32_LSI_ENABLED	1
+/* the STM32_LSI_FREQ is given by the LSI or LSI1 RC oscillator, even if not enabled */
+#if DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
 #define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi), clock_frequency)
+#elif DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
+#define STM32_LSI_FREQ		DT_PROP(DT_NODELABEL(clk_lsi1), clock_frequency)
 #else
+/* LSI clock freq should be in device tree, not from stm32Cube LSI_VALUE */
 #define STM32_LSI_FREQ		0
```

Comment:
Frequency of `0` is not a safe default and will now lead to a division by zero (hard fault) in `drivers/watchdog/wdt_iwdg_stm32.c`.

Reply:
In this case the freq could be the stm32 HAL one 
```
#warning "LSI clock freq should be in device tree, not from stm32Cube"
#define STM32_LSI_FREQ		LSI_VALUE
```